Tests/SymbolKitTests/SymbolGraph/SemanticVersion/PrecedenceTests.swift

```diff
@@ -0,0 +1,254 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import XCTest
+@testable import SymbolKit
+
+final class PrecedenceTests: XCTestCase {
+    func assert(_ version1: SymbolGraph.SemanticVersion, precedes version2: SymbolGraph.SemanticVersion) {
+        XCTAssertLessThan(version1, version2)
+        XCTAssertLessThanOrEqual(version1, version2)
```

Comment:
I think they're actually necessary. Although `Comparable` requires that all conformances satisfy certain binary relations, it has no vehicle for enforcement. So, it's completely reliant on the conformances themselves being tested on satisfaction of these relations. In fact, as shown in [SR-14665](https://github.com/apple/swift/issues/57016), in cases where comparison is not member-wise, the synthesized implementations can break the relations by themselves.

Reply:
I see. That would be helpful to information to add in a comment in these test helpers so that other developers who read the code know why it's making all these seemingly redundant assertions.

Reply:
I added a bit of hints in 98a49de378d5fdb26b56de9d927eea42219ddf43, but I think I can elaborate more in there.

Reply:
That said, I'm torn on the Comparable conformance (and to a smaller extent the Equatable conformance). I'm not against it but I'm not for it either. 

I definitely see the benefits of using these protocols when the developer's understanding of the behavior matches the actual behavior but if developer's expect two versions with different build metadata to not be equal or expect different builds in the same pre-release to be sorted by build number then that incorrect assumption could result in other bugs. 

It becomes a tradeoff between making it easy to do the right thing and making it hard to do the wrong thing. Some factors to consider would be how likely the "wrong thing" is, how bad its impact is, and how hard it's to debug.

It may just be my history with writing Objective-C code but I don't feel like function such as
`func precedes(_ other: SemanticVersion) -> Bool` or `func isEarlierRelease(comparedTo other: SemanticVersion) -> Bool` are too bad. 

I wonder what would be a good terminology to use for two versions with the same major, minor, patch, and pre-release  components but different build metadata. For example `1.0.0-alpha+001` and `1.0.0-alpha+002`. 

Would it be correct to say that they are the same "release" or would that term not include the pre-release component? If the pre-release component is considered part of the "release" then that terminology could be used in function names such as
```
func isSameRelease(as other: SemanticVersion) -> Bool { ... }
func isEarlierRelease(comparedTo other: SemanticVersion) -> Bool { ... }
```